app/src/main/java/me/quadphase/qpdex/databaseAccess/PokemonFactory.java

```diff
@@ -257,24 +163,225 @@ public static PokemonFactory getPokemonFactory(Context context) {
      * @return List of all main (no suffix) minimalPokemon in the database
      */
     public MinimalPokemon[] getAllMinimalPokemon() {
-        //TODO: Perform aggressive optimizations when possible
 
-        if (allMinimalPokemon==null) {
+        if (allMinimalPokemon==null || uniqueToNationalMap==null) {
+
             allMinimalPokemon = new MinimalPokemon[getMaxNationalID()+1];
+            uniqueToNationalMap = new HashMap<>();
 
             //Secretly added the fail-safe Pokemon...
             allMinimalPokemon[0] = PokedexManager.getInstance().missingNo.minimal();
+            uniqueToNationalMap.put(0,0);
+
+
+            //SECTION I: INITIAL QUERIES
+            //5 Cursors initialized after executing 8 queries
+
+            final Cursor mapCursor = database.rawQuery(String.format("SELECT * FROM %s WHERE %s NOT IN (SELECT %s FROM %s)",
```

Comment:
I'm guessing that Android has some optimizations for its normal query function, while the rawQuery ones are impossible to translate into said format because they do a double query.